### PR TITLE
fix(datadog): moved datadog credentials from query params to headers so that they are not logged

### PR DESCRIPTION
**Pull request by fieldju**
Fixes https://github.com/spinnaker/kayenta/issues/684

Wrote a test to reproduce the issue.

Datadog supports credentials as headers which seems to fix the broken tdd unit test.
image

Please note however since there are no Datadog integration tests that I have not tested this against the actual Datadog API.

CC: @DataDog @tylerbenson

### CODE REVIEW

File: kayenta-datadog/src/test/java/com/netflix/kayenta/datadog/functional/DatadogSecretsDoNotLeakWhenApiCalledFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2020 Netflix, Inc.
```
Comment (duftler) [inline kayenta-datadog/src/test/java/com/netflix/kayenta/datadog/functional/DatadogSecretsDoNotLeakWhenApiCalledFunctionalTest.java:2]: s/Netflix/Armory

**Comment by fieldju** [comment]
@Mergifyio backport release-1.18.x

**Comment by fieldju** [comment]
@Mergifyio backport release-1.19.x

**Comment by fieldju** [comment]
@Mergifyio backport release-1.20.x

**Pull request merged**